packages/flutter/lib/src/widgets/single_child_scroll_view.dart

```diff
@@ -0,0 +1,313 @@
+    if (_offset.pixels != value.pixels)
+      markNeedsLayout();
+    _offset = value;
+    if (attached)
+      _offset.addListener(markNeedsLayout);
+    // If we already have a size, then we should re-report the dimensions
+    // to the new ViewportOffset. If we don't then we'll report them when
+    // we establish the dimensions later, so don't worry about it now.
+    if (hasSize) {
+      assert(_minScrollExtent != null);
+      assert(_maxScrollExtent != null);
+      assert(_effectiveExtent != null);
+      offset.applyViewportDimension(_effectiveExtent);
+      if (offset.applyContentDimensions(_minScrollExtent, _maxScrollExtent))
+        markNeedsPaint();
+    }
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _offset.addListener(markNeedsPaint);
+  }
+
+  @override
+  void detach() {
+    _offset.removeListener(markNeedsPaint);
+    super.detach();
+  }
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  double get _effectiveExtent {
+    assert(hasSize);
+    switch (axis) {
+      case Axis.vertical:
+        return size.height;
+      case Axis.horizontal:
+        return size.width;
+    }
+    return null;
+  }
+
+
+  double get _minScrollExtent {
+    assert(hasSize);
+    return 0.0;
+  }
+
+  double get _maxScrollExtent {
+    assert(hasSize);
+    if (child == null)
+      return 0.0;
+    return math.max(0.0, child.size.height - size.height);
+  }
+
+  BoxConstraints _getInnerConstraints(BoxConstraints constraints) {
+    switch (axis) {
+      case Axis.horizontal:
+        return constraints.heightConstraints();
+      case Axis.vertical:
+        return constraints.widthConstraints();
+    }
+    return null;
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    if (child != null)
+      return child.getMinIntrinsicWidth(height);
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    if (child != null)
+      return child.getMaxIntrinsicWidth(height);
+    return 0.0;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    if (child != null)
+      return child.getMinIntrinsicHeight(width);
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    if (child != null)
+      return child.getMaxIntrinsicHeight(width);
+    return 0.0;
+  }
+
+  // We don't override computeDistanceToActualBaseline(), because we
+  // want the default behavior (returning null). Otherwise, as you
+  // scroll, it would shift in its parent if the parent was baseline-aligned,
+  // which makes no sense.
+
+  @override
+  void performLayout() {
+    if (child == null) {
+      size = constraints.smallest;
+    } else {
+      child.layout(_getInnerConstraints(constraints), parentUsesSize: true);
+      size = constraints.constrain(child.size);
+      final BoxParentData childParentData = child.parentData;
+      childParentData.offset = Offset.zero;
```

Comment:
I would use the RenderProxyBox trick of not using the parent data at all (and making it not be BoxParentData).